src/modules/item/NavigationButton.tsx

```diff
@@ -0,0 +1,97 @@
+import { useParams } from 'react-router';
+
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+import { AppBar, Box, Toolbar } from '@mui/material';
+
+import { ActionTriggers, DiscriminatedItem, ItemType } from '@graasp/sdk';
+import { Button } from '@graasp/ui';
+
+import { hooks, mutations } from '@/config/queryClient';
+import { useItemContext } from '@/contexts/ItemContext';
+
+const NavigationButton = ({
+  item,
+}: {
+  item: DiscriminatedItem;
+}): JSX.Element | null => {
+  const { rootId } = useParams();
+  const { setFocusedItemId } = useItemContext();
+  const { mutate: triggerAction } = mutations.usePostItemAction();
+  const { data: rootItem } = hooks.useItem(rootId);
+
+  const { data: descendants, isLoading } = hooks.useDescendants({
+    // not correct but enabled
+    id: rootId ?? '',
+    enabled: Boolean(rootId),
+  });
+
+  if (isLoading) {
+    return null;
+  }
+
+  const prevRoot: DiscriminatedItem | null = rootItem || null;
+  let prev: DiscriminatedItem | null = null;
+  let next: DiscriminatedItem | null = null;
+
+  const folderHierarchy: DiscriminatedItem[] | undefined = descendants?.filter(
+    ({ type }) => type === ItemType.FOLDER,
+  );
+
```

Comment:
I think your code would be easier if you add a `if(descendants)` above.

Reply:
Hmmm. Not really sure what you mean @pyphilia...

Reply:
@pyphilia, but are we sure that if `descendants` is truthy, it will be an array?